### ISSUE TITLE
CRD Status: Implement IPAddressPoolStatus

### ISSUE THREAD

**Issue by oribon** [metallb/metallb#2155]
### Is your feature request related to a problem?

https://github.com/metallb/metallb/blob/main/design/crd-status.md

### Describe the solution you'd like

As described in https://github.com/metallb/metallb/blob/main/design/crd-status.md#ipaddresspool-status we want to implement the status for IPAddressPools.

### Additional context

This can be used as reference: https://github.com/metallb/metallb/pull/1967

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

**Comment by cyclinder** [comment]
Hi @oribon, Can I pick this up?

**Comment by oribon** [comment]
sure! assigned, let me know if you need anything :)

**Comment by cyclinder** [comment]
thanks @oribon, I haven't started working on this yet. I may have some questions when I start working on it :)

**Comment by fedepaol** [comment]
@cyclinder as always no pressure, just checking if you are still interested in working on this.

**Comment by cyclinder** [comment]
Yes, I've been trying to understand https://github.com/metallb/metallb/pull/1967 lately, and after that, I'll start opening a pr for this.

**Comment by fedepaol** [comment]
> Yes, I've been trying to understand #1967 lately, and after that, I'll start opening a pr for this.

I'd try also to look at what we did in https://github.com/metallb/metallb/pull/2198

**Comment by oribon** [comment]
hi @cyclinder, hope everything is good
since this got a bit of priority and there wasn't activity in the past few months I'm assigning this to myself
thanks for the good intentions and please feel free to take something else if you'd like :)
